integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystemOpts.java

```diff
@@ -0,0 +1,259 @@
+    mFuseAuthPolicyCustomUser = Preconditions.checkNotNull(fuseAuthPolicyCustomUser,
+        "Option fuseAuthPolicyCustomUser for an AlluxioFuse filesystem should not be null.");
+    mFuseMaxPathCached = fuseMaxPathCached;
+    mFuseUmountTimeout = fuseUmountTimeout;
+    mIsDebug = isDebug;
+    mLibfuseOptions = Preconditions.checkNotNull(libfuseOptions,
+        "Option libfuseOptions for an AlluxioFuse filesystem should not be null.");
+    mMetadataCacheEnabled = metaDataCacheEnabled;
+    mMountPoint = Preconditions.checkNotNull(mountPoint,
+        "Option mountPoint for an AlluxioFuse filesystem should not be null.");
+    mSpecialCommandEnabled = specialCommandEnabled;
+    mStatCacheTimeout = statCacheTimeout;
+    mUserGroupTranslationEnabled = userGroupTranslationEnabled;
+  }
+
+  /**
+   * @return the alluxio path
+   */
+  public String getAlluxioPath() {
+    return mAlluxioPath;
+  }
+
+  /**
+   * @return the Filesystem name
+   */
+  public String getFsName() {
+    return mFsName;
+  }
+
+  /**
+   * @return the authorization policy class for Fuse
+   */
+  public Class getFuseAuthPolicyClass() {
+    return mFuseAuthPolicyClass;
+  }
+
+  /**
+   * @return the fuse group for the authorization policy
+   */
+  public Optional<String> getFuseAuthPolicyCustomGroup() {
+    return mFuseAuthPolicyCustomGroup;
+  }
+
+  /**
+   * @return the fuse user for the authorization policy
+   */
+  public Optional<String> getFuseAuthPolicyCustomUser() {
+    return mFuseAuthPolicyCustomUser;
+  }
+
+  /**
+   * @return the max number of FUSE-to-Alluxio path mappings to cache
+   */
+  public int getFuseMaxPathCached() {
+    return mFuseMaxPathCached;
+  }
+
+  /**
+   * @return the wait timeout for fuse to umount
+   */
+  public int getFuseUmountTimeout() {
+    return mFuseUmountTimeout;
+  }
+
+  /**
+   * @return if using debug-level logging for AlluxiJniFuseFileSystem and Libfuse
+   */
+  public boolean isDebug() {
+    return mIsDebug;
+  }
+
+  /**
+   * @return options for Libfuse
+   */
+  public List<String> getFuseOptions() {
+    return mLibfuseOptions;
+  }
+
+  /**
+   * @return if caching metadata on client side
+   */
+  public boolean isMetadataCacheEnabled() {
+    return mMetadataCacheEnabled;
+  }
+
+  /**
+   * @return the mount point
+   */
+  public String getMountPoint() {
+    return mMountPoint;
+  }
+
+  /**
+   * @return whether AlluxioFuse special command is enabled
+   */
+  public boolean isSpecialCommandEnabled() {
+    return mSpecialCommandEnabled;
+  }
+
+  /**
+   * @return the time the result of statfs is cached
+   */
+  public long getStatCacheTimeout() {
+    return mStatCacheTimeout;
+  }
+
+  /**
+   * @return whether user and group are translated from Alluxio to Unix
+   */
+  public boolean isUserGroupTranslationEnabled() {
+    return mUserGroupTranslationEnabled;
+  }
+
+  private static List<String> optimizeAndFormatFuseOptions(List<String> fuseOpts) {
```

Comment:
This looks more like a util function
Options usually is initialize once and read only, other util methods should not contain in the options class

Reply:
This is a process which we must do before passing the libfuse options (allow_other, big_write, etc.) to libfuse, so I don't think it's a util function. 
After reading the libfuse options from configuration or command line, we have to process them. This is before we create the read-only AlluxioFuseFileSystemOpts object

Reply:
@rongrong is it a common practice of adding options-related processing methods in the options class?